### PR TITLE
Add Draconicevolution energy core block compat

### PR DESCRIPTION
**Pull request by brachy84**
:)

### CODE REVIEW

File: src/main/java/com/cleanroommc/groovyscript/core/mixin/draconicevolution/EnergyCoreStructureAccessor.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.cleanroommc.groovyscript.core.mixin.draconicevolution;
+
+import com.brandon3055.brandonscore.lib.MultiBlockStorage;
+import com.brandon3055.draconicevolution.world.EnergyCoreStructure;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.gen.Accessor;
+
+@Mixin(EnergyCoreStructure.class)
```
Comment (WaitingIdly) [inline src/main/java/com/cleanroommc/groovyscript/core/mixin/draconicevolution/EnergyCoreStructureAccessor.java:8]: should be `remap = false`, correct?

File: src/main/java/com/cleanroommc/groovyscript/core/mixin/draconicevolution/MultiBlockStorageAccessor.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.cleanroommc.groovyscript.core.mixin.draconicevolution;
+
+import com.brandon3055.brandonscore.lib.MultiBlockStorage;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.gen.Accessor;
+
+@Mixin(MultiBlockStorage.class)
```
Comment (WaitingIdly) [inline src/main/java/com/cleanroommc/groovyscript/core/mixin/draconicevolution/MultiBlockStorageAccessor.java:7]: remap?

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/draconicevolution/EnergyCore.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        onReload(); // increases version to 1
+    }
+
+    @Override
+    public Collection<String> getAliases() {
+        return Alias.generateOfClass(EnergyCore.class);
+    }
+
+    @Override
+    public void onReload() {
```
Comment (WaitingIdly) [inline src/main/java/com/cleanroommc/groovyscript/compat/mods/draconicevolution/EnergyCore.java:52]: the video uploaded on discord as a showcase clearly shows reloading functioning, but i cannot reproduce this behavior. on my computer, after a change reloading does not cause the multiblock to become invalidated, and requires a full restart to see the affect.

File: src/main/resources/assets/groovyscript/lang/en_us.lang [hunk 4]
```diff
@@ -624,6 +624,10 @@ groovyscript.wiki.compactmachines3.miniaturization.keyMap.value=Sets the IBlockS
 groovyscript.wiki.compactmachines3.miniaturization.symmetrical.value=Sets if the recipe does not have to test all 4 rotations to determine if the multiblock is valid
 
 # Draconic Evolution
+groovyscript.wiki.draconicevolution.energy_core.title=Energy Core
+groovyscript.wiki.draconicevolution.energy_core.description=A multiblock which stores energy
```
Comment (WaitingIdly) [inline src/main/resources/assets/groovyscript/lang/en_us.lang:628]: would like a more detailed description, maybe like
```suggestion
groovyscript.wiki.draconicevolution.energy_core.description=A multiblock with 8 tiers for storing large amounts of energy.
```

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/draconicevolution/helpers/BlockStates.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        return false;
+    }
+
+    public static ItemStack transformStateToStack(IBlockState state) {
+        return new ItemStack(state.getBlock(), 1, state.getBlock().getMetaFromState(state));
+    }
+
+    @SuppressWarnings("deprecation")
+    public static IBlockState transformStackToState(ItemStack stack) {
+        if (!(stack.getItem() instanceof ItemBlock block))
```
Comment (WaitingIdly) [inline src/main/java/com/cleanroommc/groovyscript/compat/mods/draconicevolution/helpers/BlockStates.java:104]: i think the logic on this is better *not* inverted, way more readable
```
        if (stack.getItem() instanceof ItemBlock block) {
            return block.getBlock().getStateFromMeta(stack.getMetadata());
        }
        throw new IllegalArgumentException("[LabsDraconicEvolution] Stack's Item must extend ItemBlock!");
```

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/draconicevolution/helpers/BlockStates.java [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+            throw new IllegalArgumentException("[LabsDraconicEvolution] Stack's Item must extend ItemBlock!");
```
Comment (WaitingIdly) [inline src/main/java/com/cleanroommc/groovyscript/compat/mods/draconicevolution/helpers/BlockStates.java:105]: er, `LabsDraconicEvolution`?

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/draconicevolution/helpers/BlockStateEnergyCoreStructure.java [hunk 7]
```diff
@@ -0,0 +1,777 @@
...
+        storage.addRow(e, e, e, e, D, D, D, e, e, e, e);
+
+        storage.mirrorHalf();
+
+        return storage;
+    }
+
+    private BlockStateMultiblockStorage buildTierOMG() {
+        BlockStateMultiblockStorage storage = new BlockStateMultiblockStorage(13, helper, this);
+        BlockStates e = BlockStates.ANY, X = coreBlock(), R = BlockStates.redstone(), D = BlockStates.draconium(), A = BlockStates.draconic();
```
Comment (WaitingIdly) [inline src/main/java/com/cleanroommc/groovyscript/compat/mods/draconicevolution/helpers/BlockStateEnergyCoreStructure.java:562]: i think new variable declarations belong on different lines

**Review by WaitingIdly — APPROVED** [review]
it reloads properly now, just have a few suggestions relating to code style/some text.

**Comment by IntegerLimit** [comment]
Please add config.

**Review by WaitingIdly — APPROVED** [review]

**Pull request merged**